app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/LibreUtils.java

```diff
@@ -111,7 +108,13 @@ public static boolean isSensorReady(byte sensorStatusByte) {
     
         Log.i(TAG, "Sensor status is: " + sensorStatusString);
     
-        if (testWithDeadSensor) return true;
+        
+        
+        
+        if (Pref.getBooleanDefaultFalse("allow_testing_with_dead_sensor")) {
```

Comment:
I think this should be in a public method, eg `boolean getTestWithDeadSensor()` so that it matches the removed getter and can be accessed externally by other things which wish to check this flag, then replace this code block with a call to that method and then the resulting logic for logging and returning.

Then update the method so it definitely requires engineering mode as well with something like `&& Pref.getBooleanDefaultFalse("engineering_mode")`

Then of course mirror the same for wear. Hope that makes sense.